src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3938,6 +3940,9 @@ def _check_running(self):
             else:
                 self._prof.prof('exec', msg='execution complete', uid=cu['_id'])
 
+                # make sure proc is collected
+                cu['proc'].wait()
```

Comment:
If this can actually be required after a successful poll then I'm fine with keeping it.